## CAPSTONE

# Data analysis road map


1. Load in the data
2. Clean data
    - Check for nulls
    - Unpack any
3. 

In [1]:
import json
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


In [2]:
import json
mag_sub = []
for line in open('Magazine_Subscriptions.json','r'):
    mag_sub.append(json.loads(line))


In [3]:
import json
meta_data = []
for line in open('meta_Magazine_Subscriptions.json','r'):
    meta_data.append(json.loads(line))


In [4]:
mag_sub[0]

{'overall': 5.0,
 'vote': '9',
 'verified': False,
 'reviewTime': '11 8, 2001',
 'reviewerID': 'AH2IFH762VY5U',
 'asin': 'B00005N7P0',
 'reviewerName': 'ted sedlmayr',
 'reviewText': 'for computer enthusiast, MaxPC is a welcome sight in your mailbox. i can remember for years savorying every page of "boot" (as it was called in beginning) as i was (and still am) obcessed with PC\'s. Anyone, from advanced users - to beginners looking for knowledge - can profit from every issue of MaxPC. the icing on the cake is the subscription that comes with a CD-ROM as it is packed with demos, utilities, and other useful apps (very helpful for those not blessed with broadband connections). Until I discovered the community of hardware enthusiast web sites, MaxPC, formerly "boot", was my only really informative source for computing news and articles. To this day, i consider my subscription to it worth more than 10 subscriptions to most other computing mags. I can\'t wait until they merge with DVD media a

In [5]:
len(mag_sub)

89689

In [6]:
mag_df = pd.DataFrame(mag_sub)
meta_df = pd.DataFrame(meta_data)

In [7]:
mag_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89689 entries, 0 to 89688
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   overall         89689 non-null  float64
 1   vote            24103 non-null  object 
 2   verified        89689 non-null  bool   
 3   reviewTime      89689 non-null  object 
 4   reviewerID      89689 non-null  object 
 5   asin            89689 non-null  object 
 6   reviewerName    89687 non-null  object 
 7   reviewText      89656 non-null  object 
 8   summary         89670 non-null  object 
 9   unixReviewTime  89689 non-null  int64  
 10  style           51398 non-null  object 
 11  image           135 non-null    object 
dtypes: bool(1), float64(1), int64(1), object(9)
memory usage: 7.6+ MB


In [8]:
meta_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3385 entries, 0 to 3384
Data columns (total 19 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   category         3385 non-null   object
 1   tech1            3385 non-null   object
 2   description      3385 non-null   object
 3   fit              3385 non-null   object
 4   title            3385 non-null   object
 5   also_buy         3385 non-null   object
 6   tech2            3385 non-null   object
 7   brand            3385 non-null   object
 8   feature          3385 non-null   object
 9   rank             3385 non-null   object
 10  also_view        3385 non-null   object
 11  details          3385 non-null   object
 12  main_cat         3385 non-null   object
 13  similar_item     3385 non-null   object
 14  date             3385 non-null   object
 15  price            3385 non-null   object
 16  asin             3385 non-null   object
 17  imageURL         3385 non-null   

In [9]:
mag_df.head()

overall vote  verified   reviewTime      reviewerID        asin  \
0      5.0    9     False   11 8, 2001   AH2IFH762VY5U  B00005N7P0   
1      5.0    9     False  10 31, 2001   AOSFI0JEYU4XM  B00005N7P0   
2      3.0   14     False  03 24, 2007  A3JPFWKS83R49V  B00005N7OJ   
3      5.0   13     False  11 10, 2006  A19FKU6JZQ2ECJ  B00005N7OJ   
4      5.0  NaN      True  07 14, 2014  A25MDGOMZ2GALN  B00005N7P0   

         reviewerName                                         reviewText  \
0        ted sedlmayr  for computer enthusiast, MaxPC is a welcome si...   
1     Amazon Customer  Thank god this is not a Ziff Davis publication...   
2         Bryan Carey  Antiques Magazine is a publication made for an...   
3  Patricia L. Porada  This beautiful magazine is in itself a work of...   
4               Alvey                          A great read every issue.   

                                           summary  unixReviewTime  \
0           AVID READER SINCE "boot"  WAS THE NAME      1005177600   
1                               The straight scoop      1004486400   
2  Antiques Magazine is Good, but not for Everyone      1174694400   
3                           THE  DISCERNING READER      1163116800   
4                                       Five Stars      1405296000   

                            style image  
0                             NaN   NaN  
1                             NaN   NaN  
2  {'Format:': ' Print Magazine'}   NaN  
3  {'Format:': ' Print Magazine'}   NaN  
4                             NaN   NaN

In [32]:
meta_df

category tech1  \
0     [Magazine Subscriptions, Professional & Educat...         
1     [Magazine Subscriptions, Arts, Music &amp; Pho...         
2     [Magazine Subscriptions, Fashion &amp; Style, ...         
3     [Magazine Subscriptions, Sports, Recreation & ...         
4     [Magazine Subscriptions, Professional & Educat...         
...                                                 ...   ...   
3380                                                 []         
3381  [Magazine Subscriptions, Sports, Recreation & ...         
3382          [Magazine Subscriptions, Fashion & Style]         
3383                                                 []         
3384                                                 []         

                                            description fit  \
0     [REASON is edited for people interested in eco...       
1     [Written by and for musicians. Covers a variet...       
2     [Allure is the beauty expert. Every issue is f...       
3     [FLIGHT JOURNAL includes articles on aviation ...       
4     [RIDER is published for the road and street ri...       
...                                                 ...  ..   
3380  [The game's most widely read publication, GOLF...       
3381  [Built for the true classic car enthusiast, ea...       
3382  [InStyle celebrates the private side of public...       
3383  [Sports Illustrated brings you spectacular act...       
3384  [Get fresh ideas and inspiration on how to mak...       

                                              title  \
0     <span class="a-size-medium a-color-secondary"   
1     <span class="a-size-medium a-color-secondary"   
2     <span class="a-size-medium a-color-secondary"   
3     <span class="a-size-medium a-color-secondary"   
4     <span class="a-size-medium a-color-secondary"   
...                                             ...   
3380  <span class="a-size-medium a-color-secondary"   
3381  <span class="a-size-medium a-color-secondary"   
3382  <span class="a-size-medium a-color-secondary"   
3383  <span class="a-size-medium a-color-secondary"   
3384  <span class="a-size-medium a-color-secondary"   

                                               also_buy tech2  \
0     [B002PXVYLE, B01MCU84LB, B000UHI2LW, B01AKS14A...         
1     [B002PXVYGE, B0054LRNC8, B000BVEELE, B00006KC3...         
2     [B001THPA4O, B002PXVZWW, B001THPA1M, B001THPA1...         
3     [B07JVF7QW4, B00ATQ6FPY, B002G551F6, B00008CGW...         
4     [B002PXVYD2, B01BM7TOU6, B000060MKJ, B000BNNIG...         
...                                                 ...   ...   
3380                                                 []         
3381  [B002PXVYO6, B01FIR6AJ4, B00006KGT1, B01DPP7H5...         
3382                                                 []         
3383                                                 []         
3384                                                 []         

                                  brand feature  \
0                       Reason Magazine      []   
1              String Letter Publishers      []   
2               Conde Nast Publications      []   
3                     AirAge Publishing      []   
4     EPG Media & Specialty Information      []   
...                                 ...     ...   
3380                   TI Golf Holdings      []   
3381         Motorsport Marketing, Inc.      []   
3382               Meredith Corporation      []   
3383               Meredith Corporation      []   
3384               Meredith Corporation      []   

                                 rank  \
0                                  []   
1     742 in Magazine Subscriptions (   
2                                  []   
3                                  []   
4                                  []   
...                               ...   
3380                               []   
3381                               []   
3382                               []   
3383                               []   
338

In [14]:
meta_df['details'][0]

{'Format:': 'Print Magazine',
 'Shipping: ': 'Currently, item can be shipped only within the U.S.',
 'Publisher:': 'Reason Magazine',
 'ASIN: ': 'B00005N7NQ'}

Check the number of unique products in the database

In [13]:
mag_df['style'].value_counts()

{'Format:': ' Print Magazine'}    37638
{'Format:': ' Kindle Edition'}    13760
Name: style, dtype: int64

In [34]:
len(pd.unique(mag_df['reviewerID']))

72098

There are 2428 unique products out of 89689 data entries


In [13]:
len(pd.unique(meta_df['asin']))

2320

In [19]:
# check mean, std entries for each item
mag_df.describe()

overall  unixReviewTime
count  89689.000000    8.968900e+04
mean       4.036638    1.389447e+09
std        1.419791    1.039430e+08
min        1.000000    1.004054e+09
25%        3.000000    1.355184e+09
50%        5.000000    1.414109e+09
75%        5.000000    1.461542e+09
max        5.000000    1.537315e+09

The item with the most entries has 163 entries and every item has atleast 5 reviews. The mean is about 11 entries and the std is about 13 entries.